packages/deprecation-crawler/src/lib/utils.ts

```diff
@@ -46,60 +46,101 @@ export function readFile(path: string) {
   return '';
 }
 
-export function concat<I>(
-  processes: CrawlerProcess<I, I>[]
-): CrawlerProcess<I, I> {
-  return async function (d: I): Promise<I> {
+export function run(
+  tasks: ((config: CrawlConfig) => CrawlerProcess)[],
+  config: CrawlConfig
+): CrawlerProcess {
+  return concat(tasks.map((s) => s(config)));
+}
+
+export function concat(processes: CrawlerProcess[]): CrawlerProcess {
+  return async function (d: CrawledRelease): Promise<CrawledRelease | void> {
     return await processes.reduce(
       async (deps, processor) => await processor(await deps),
       Promise.resolve(d)
     );
   };
 }
 
-export function tap<I>(
-  process: CrawlerProcess<I, I | void>
-): CrawlerProcess<I, I> {
-  return async function (d: I): Promise<I> {
+export function tap(process: CrawlerProcess): CrawlerProcess {
+  return async function (d: CrawledRelease): Promise<CrawledRelease> {
     await process(d);
     return Promise.resolve(d);
   };
 }
 
-export function askToSkip<I>(
+export function askToSkip(
   question: string,
-  crawlerProcess: CrawlerProcess<I, I>
-): CrawlerProcess<I, I> {
-  return async function (d: I): Promise<I> {
-    if (sandBoxMode()) {
-      return await crawlerProcess(d);
-    }
+  crawlerProcess: CrawlerProcess,
+  options: {
+    precondition?: (r: CrawledRelease) => Promise<boolean>;
+  } = {
+    precondition: async () => true,
+  }
+): CrawlerProcess {
+  return async function (d: CrawledRelease): Promise<CrawledRelease | void> {
+    const isPreconditionMet = await options.precondition(d);
 
-    const answer: { proceed: boolean } = await prompt([
-      {
-        type: 'confirm',
-        name: 'proceed',
-        message: question,
-        initial: true,
-      },
-    ]);
+    if (!isPreconditionMet) {
```

Comment:
Ignore this step (and don't ask the question) when the preconditions have not been met.